crates/e2e/tests/e2e/limit_orders.rs

```diff
@@ -446,3 +457,100 @@ async fn too_many_limit_orders_test(web3: Web3) {
     assert_eq!(status, 400);
     assert!(body.contains("TooManyLimitOrders"));
 }
+
+async fn forked_single_limit_order_test(web3: Web3) {
+    let mut onchain = OnchainComponents::at(web3.clone()).await;
+    let forked_node_api = web3.api::<ForkedNodeApi<_>>();
+
+    let auth_manager = onchain
+        .contracts()
+        .gp_authenticator
+        .manager()
+        .call()
+        .await
+        .unwrap();
+
+    forked_node_api.impersonate(&auth_manager).await.unwrap();
+
+    let [solver] = onchain
+        .make_solvers_forked(to_wei(1), ethcontract::Account::Local(auth_manager, None))
+        .await;
+
+    let [trader] = onchain.make_accounts(to_wei(1)).await;
+
+    let token_usdc = ERC20::at(
+        &web3,
+        "0xa0b86991c6218b36c1d19d4a2e9eb0ce3606eb48"
+            .parse()
+            .unwrap(),
+    );
+
+    let token_usdt = ERC20::at(
+        &web3,
+        "0xdac17f958d2ee523a2206206994597c13d831ec7"
+            .parse()
+            .unwrap(),
+    );
+
+    // Give trader some USDC
+    let finder = TokenOwnerFinder::from_web3(web3.clone()).await;
+    forked_node_api
+        .set_erc20_balance(trader.address(), &token_usdc, to_mwei(1000), finder)
+        .await
+        .unwrap();
```

Comment:
I'm not super versed in testing using forks but whenever I saw it the test aimed to take some state that existed on a chain at some specific block and use it as is (whenever possible).
That way you wouldn't run into issues where some test starts to fail all of a sudden because something changed.
This would also allow you to run the CI job unconditionally.

There are also tokens where overwriting the balance can be finicky (I think stETH) is one of them. Not requiring the overwrite in the first place means you can test with those unusual tokens as well.

Also you wouldn't have to depend on the token owner logic. (I think this will be dropped long term)

Reply:
Yeah that's a good point, I'll change it to pin to a specific block and use some known whale addresses as per that block.

Reply:
Hmm, one thing I didn't consider is that this now requires an archive node. Let's see if we have one of those kicking around. 👀

Reply:
That's true, but many service providers offer a generous free tier with archive nodes these days, and there are not actually too many RPC calls happening for now.